### PR TITLE
various compiler fixes

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

**Comment by vikerman** [comment]
Fixes #8336

**Comment by tbosch** [comment]
Reviewed in person with @mhevery

File: modules/angular2/test/compiler/template_parser_spec.ts [hunk 1]
```diff
@@ -1139,6 +1147,14 @@ There is no directive with "exportAs" set to "dirA" ("<div [ERROR ->]#a="dirA"><
               .toEqual([['div', null], ['template', 1], ['a', null]]);
         });
       });
+
+      it('should support other directives before the component', () => {
+        expect(humanizeContentProjection(
+                   parse('<div><a *ngIf="cond" ngProjectAs="b"></a></div>',
```
Comment (tbosch) [inline modules/angular2/test/compiler/template_parser_spec.ts]: Make this simpler

File: modules/angular2/test/core/linker/regression_integration_spec.ts [hunk 2]
```diff
@@ -187,3 +204,8 @@ class PlatformPipe implements PipeTransform {
 class CustomPipe implements PipeTransform {
   transform(value: any): any { return 'someCustomPipe'; }
 }
+
```
Comment (tbosch) [inline modules/angular2/test/core/linker/regression_integration_spec.ts]: Remove this

**Pull request merged**